src/foam/nanos/ruler/Rule.js

```diff
@@ -79,6 +80,7 @@
       class: 'FObjectProperty',
       of: 'foam.mlang.predicate.Predicate',
       name: 'predicate',
+      storageTransient: true,
```

Comment:
Why is this storage transient?

Reply:
Neither action nor predicate implements Comparable. The tests run fine because we mock daos with MDAO but if you switch to JDAO(what we need) it throws exception.

Reply:
If they are modelled, rather than just java implementations, then they should work fine.

Reply:
do you want me to model them as part of this pr?

Reply:
Can be done in follow up PR.